Data Quality dashboard (#3059)
elastic/security-docs: docs/dashboards/data-quality-dashboard.asciidoc

.Requirements
[sidebar]
--
To use the Data Quality dashboard, you need at least the following {ref}/security-privileges.html#privileges-list-indices[privileges] for each index you want to check:

* `monitor` or `manage` (required for the {ref}/indices-stats.html[Index stats API])
* `view_index_metadata` or `manage_ilm` (required for the {ref}/ilm-explain-lifecycle.html[Explain lifecycle API])
* `view_index_metadata` or `manage` (required for the {ref}/indices-get-mapping.html[Get mapping API])
* `read` (required for the {ref}/search-search.html[Search API])
--

[discrete]
== Check indices
When you open the dashboard, data does not appear until you select indices to check.

* Check multiple indices: To check all indices in the current data view, click *Check all* at the top of the dashboard. A progress indicator will appear.
* Check a single index: To check a single index, expand it using the arrow on the left. Checking a single index is faster than checking all indices.

[discrete]
== Learn more about checked index fields

> After an index is checked, either an X (❌) or a checkmark (✅) appears in its *Result* column. The X (❌) indicates mapping problems in an index. To view index details, including which fields weren't successfully mapped, click the arrow next to the result.

Reviewer: ```suggestion
After an index is checked, an X (❌) or a checkmark (✅) appears in its *Result* column. The X (❌) indicates mapping problems in an index. To view index details, including which fields weren't successfully mapped, click the arrow next to the result.
```

[role="screenshot"]
image::images/data-qual-dash-detail.png[An expanded index with some failed results in the Data Quality dashboard]

[discrete]
== Track unsuccessful mappings

You can share data quality results to help track your team's remediation tasks. First, follow the instructions under *Check indices* to generate results, then either:

*Share results for all indices in the current data view:*

. Click *Take action* at the top of the dashboard.
. Select *Add to new case* to open a new <<cases-overview,case>>, or click *Copy to clipboard* to copy a Markdown report to your clipboard.

*Share results for one index:*

. Expand an index by clicking the arrow to the left of its *Result*.
. From the *Summary* or *Incompatible fields* tabs, select *Add to new case* to open a new <<cases-overview,case>>, or click *Copy to clipboard* to copy a Markdown report to your clipboard.

> 

Reviewer: An image depicting one of these examples would be nice here.

[[data-quality-dash]]
= Data Quality dashboard

> The Data Quality dashboard shows you whether your data is effectively mapped to the https://www.elastic.co/guide/en/ecs/current/ecs-reference.html[Elastic Common Schema] (ECS). Successful {ref}/mapping.html[mapping] enables you to search, visualize, and interact with your data throughout {es-sec} and {kib}.

Reviewer 2: ```suggestion
The Data Quality dashboard shows you whether your data is correctly mapped to the https://www.elastic.co/guide/en/ecs/current/ecs-reference.html[Elastic Common Schema] (ECS). Successful {ref}/mapping.html[mapping] enables you to search, visualize, and interact with your data throughout {es-sec} and {kib}.
```

Reviewer 2: Both likely work but I feel like "correctly" is the slightly more accurate term to use. If I'm understanding this dashboard's purpose, it's to help users determine whether their data is ECS-_compliant_, meaning their data needs to fit into a defined set of rules. WDYT @andrew-goldstein @michaelolo24? Does my suggested change create too much of a closed statement?

image::images/data-qual-dash.png[The Data Quality dashboard]

You can use the Data Quality dashboard to:

> * Check one or multiple indices (by default, the indices used by {es-sec}) for unsuccessful mappings, to help you identify problems.

Reviewer 2: ```suggestion
* Check one or multiple indices for unsuccessful mappings, to help you identify problems. By default, the indices used by {es-sec} are checked for issues.
```

* View detailed information about the fields in checked indices.

> * Track unsuccessful mappings by creating a case or markdown report based on Data Quality results.

Reviewer 2: ```suggestion
* Track unsuccessful mappings by creating a case or Markdown report based on data quality results.
```

Reviewer 2: I'm not sure if you're referring to the dashboard results here, so feel free to reject this change if you're using the short-hand version of the dashboard name.

> NOTE: The Data Quality dashboard does not show data from cold or frozen {ref}/data-tiers.html[data tiers]. It also does not display data from remote clusters using cross-cluster search — to view data from another cluster, log in to that cluster's {kib} instance.

Reviewer 3: Edit suggestions for tone, and a semicolon would prob be more appropriate to join clauses like these. I think a dash is better for interjecting "extra" or maybe unexpected information, not just transitioning to the next step in a procedure.
```suggestion
NOTE: The Data Quality dashboard doesn't show data from cold or frozen {ref}/data-tiers.html[data tiers]. It also doesn't display data from remote clusters using cross-cluster search; to view data from another cluster, log in to that cluster's {kib} instance.
```

> .Requirements

Reviewer 2: This information is a little hard to read in the list format. What do you think about creating a new section and moving the content into a table?

Author: I am open to it but leaning the other way. Curious to hear what Joe and Janeen think when they review.

Author: I'm open to removing the "Required for ___" bits as you suggest, but I think I like including them for any power users or wannabe-power users who are concerned with the "why". Janeen what do you think?

[sidebar]
--

> To use the Data Quality dashboard, you need at least the following {ref}/security-privileges.html#privileges-list-indices[permissions] for each index you want to check:

Reviewer 3: The linked page calls them "privileges" (and I think that's what we use in other docs too):
```suggestion
To use the Data Quality dashboard, you need at least the following {ref}/security-privileges.html#privileges-list-indices[privileges] for each index you want to check:
```

> . `monitor` or `manage` (required for the {ref}/indices-stats.html[Index stats API])
> . `view_index_metadata` or `manage_ilm` (required for the {ref}/ilm-explain-lifecycle.html[Explain lifecycle API])
> . `view_index_metadata` or `manage` (required for the {ref}/indices-get-mapping.html[Get mapping API])
> . `read` (required for the {ref}/search-search.html[Search API])

Reviewer 3: Only use numbered steps when things need to be done in a specific order; bullets are more appropriate for a list like this. I'll leave the API statements untouched for now, but see my question above about whether they're necessary.
```suggestion
* `monitor` or `manage` (required for the {ref}/indices-stats.html[Index stats API])
* `view_index_metadata` or `manage_ilm` (required for the {ref}/ilm-explain-lifecycle.html[Explain lifecycle API])
* `view_index_metadata` or `manage` (required for the {ref}/indices-get-mapping.html[Get mapping API])
* `read` (required for the {ref}/search-search.html[Search API])
```

--

[discrete]
== Check indices
When you open the dashboard, data does not appear until you select which indices to check.

*Check multiple indices:*
To check all indices in the current data view, click *Check all* at the top of the dashboard. A progress indicator will appear.

*Check a single index:*
To check a single index, expand it using the arrow on the left. Checking a single index is faster than checking all indices.

[discrete]

> == View detailed information about the fields in a checked index

Reviewer 2: You can prob shorten this since you explain it in the following sentence. Also, if you want to remove "view" from the title, here's another alternative:

```suggestion
== Learn more about checked index fields
```

> After an index is checked, either an *X* (❌) or a *checkmark* (✅) appears in its *Result* column. To view index details, including which fields were not successfully mapped, click the arrow to the left of the result.

Reviewer 2: Does the `X` mean the index has mapping problems? If yes, it could be good to include that detail in the first sentence. Maybe something like:

```suggestion
After an index is checked, either an *X* (❌) or a *checkmark* (✅) appears in its *Result* column. The *X* (❌) indicates mapping problems in an index. To view index details, including which fields were not successfully mapped, click the arrow to the left of the result.
```

image::images/data-qual-dash-detail.png[An expanded index with some failed results in the Data Quality dashboard]

[discrete]

> == Track unsuccessful mappings by creating a case or markdown report based on Data Quality results

Reviewer 2: ```suggestion
== Track unsuccessful mappings 
```

> You can share data quality results to help track your team's remediation tasks. First, follow the instructions above under *Check indices* to generate results, then either:

Reviewer 2: ```suggestion
You can share data quality results to help track your team's remediation tasks. First, follow the instructions under *Check indices* to generate results, then either:
```

*Share results for all indices in the current data view:*

. Click *Take action* at the top of the dashboard.

> . Select either *Add to new case* to open the Create new case flyout, or click *Copy to clipboard* to copy a markdown report to your clipboard.

Reviewer 2: ```suggestion
. Select *Add to new case* to open a new <<cases-overview,case>>, or click *Copy to clipboard* to copy a Markdown report to your clipboard.
```

*Share results for one index:*

> . First, expand an index by clicking the arrow to the left of its *Result*.

Reviewer 2: ```suggestion
.Expand an index by clicking the arrow to the left of its *Result*.
```

> . From the *Summary* or *Incompatible fields* tabs, select either *Add to new case* to open the Create new case flyout, or click *Copy to clipboard* to copy a markdown report to your clipboard.

Reviewer 2: ```suggestion
. From the *Summary* or *Incompatible fields* tabs, select *Add to new case* to open a new <<cases-overview,case>>, or click *Copy to clipboard* to copy a Markdown report to your clipboard.
```

> NOTE: For more information about how to remediate unsuccessful mappings, refer to {ref}/mapping.html[mapping].

Reviewer 2: ```suggestion
NOTE: For more information about how to fix mapping problems, refer to {ref}/mapping.html[mapping].
```
